packages/web/components/token-details/token-details.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from "react";
+
+import { Icon } from "~/components/assets";
+import LinkIconButton from "~/components/buttons/link-icon-button";
+
+const TEXT_CHAR_LIMIT = 485;
+
+function TokenDetails() {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const text = `Osmosis is a decentralized exchange (DEX) that uses the Cosmos SDK for optimized functionality with other blockchains. Its automated market makers set it apart, allowing enhanced market responsiveness for liquidity providers.\n\nOsmosis is a multi-chain DEX specifically built to function with other blockchains in the Cosmos ecosystem. Communication withthese other blockchains is achieved via its Inter-Blockchain Communication protocol (IBC). Because Osmosis is linked to other Tendermint blockchains, providing smooth, fast trades,`;
+
+  return (
+    <div className="flex flex-col items-start gap-3 self-stretch rounded-5xl border border-osmoverse-800 bg-osmoverse-900 p-10 xl:gap-6 1.5xs:gap-6">
+      <TokenStats />
+      <div className="flex flex-col items-start self-stretch">
+        <div className="flex flex-col items-start gap-4.5 self-stretch 1.5xs:gap-6">
+          <div className="flex items-center gap-8 1.5xs:flex-col 1.5xs:gap-4">
+            <h6 className="text-lg font-h6 leading-6 text-osmoverse-100">
+              About OSMO
+            </h6>
+            <div className="flex items-center gap-2">
+              <LinkIconButton
+                href="/"
+                mode="icon-social"
+                size="md-icon-social"
+                aria-label="View on X"
+                icon={
+                  <Icon
+                    className="h-[16px] w-[16px] text-osmoverse-400"
+                    id="X"
+                  />
+                }
+              />
+              <LinkIconButton
+                href="/"
+                mode="icon-social"
+                size="md-icon-social"
+                aria-label="View website"
+                icon={
+                  <Icon
+                    className="h-[24px] w-[24px] text-osmoverse-400"
+                    id="web"
+                  />
+                }
+              />
+              <LinkIconButton
+                href="/"
+                mode="icon-social"
+                size="md-icon-social"
+                aria-label="View on CoinGecko"
+                icon={
+                  <Icon
+                    className="h-[42px] w-[42px] text-osmoverse-300"
+                    id="coingecko"
+                  />
+                }
+              />
+            </div>
+          </div>
+          <div
+            className={`${!isExpanded && "aftershadow"} relative self-stretch`}
+          >
+            <p className="breakspaces font-base self-stretch font-subtitle1 text-osmoverse-200 transition-all">
+              {isExpanded ? text : text.substring(0, TEXT_CHAR_LIMIT)}
```

Comment:
The basic logic is right, but it does not handle the case where I have text that is shorter than the maximum number of characters, in which case there is no need to show the "show more" button.